panoptes_client/panoptes.py

```diff
@@ -231,9 +233,16 @@ def get_csrf_token(self):
 
     def get_bearer_token(self):
         if not self.bearer_token or self.bearer_expires > datetime.now():
-            if not self.logged_in:
+
+            if self.client_secret:
+                grant_type = 'client_credentials'
+            elif self.password:
+                grant_type = 'password'
+
```

Comment:
Will `grant_type` possibly be undefined if there's no client secret and no password?